toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -620,14 +620,28 @@ func InitializeSinglePartition(diskDevPath string, partitionNumber int, partitio
 
 	logger.Log.Debugf("Initializing partition device path: %v", partDevPath)
 
-	// Set partition friendly name (only for gpt)
+	// Set partition friendly name and partition type UUID (only for gpt)
 	if partitionTableType == "gpt" {
 		partitionName := partition.Name
 		_, stderr, err := shell.Execute("flock", "--timeout", timeoutInSeconds, diskDevPath, "parted", diskDevPath, "--script", "name", partitionNumberStr, partitionName)
 		if err != nil {
 			logger.Log.Warnf("Failed to set partition friendly name using parted: %v", stderr)
 			// Not-fatal
 		}
+
+		if partition.TypeUUID != "" || partition.Type != "" {
+			var typeUUID string
+			if partition.TypeUUID != "" {
+				typeUUID = partition.TypeUUID
+			} else {
+				typeUUID = configuration.PartitionTypeNameToUUID[partition.Type]
+			}
+			_, stderr, err := shell.Execute("flock", "--timeout", timeoutInSeconds, diskDevPath, "parted", diskDevPath, "--script", "type", partitionNumberStr, typeUUID)
+			if err != nil {
+				logger.Log.Warnf("failed to set partition type using parted: %v", stderr)
+				// Not-fatal
```

Comment:
If it's non-fatal, we should also `err = nil`, so that the caller doesn't accidentally crash. We should probably do the same thing in the code above, which is setting the friendly name.